.github/workflows/terraform.yaml

```diff
@@ -0,0 +1,90 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: Terraform checks
+
+on:
+  workflow_call:
+    inputs:
+      module-directory:
+        description: "Directory where the terraform module is hosted in the repo"
+        default: "./terraform"
+        required: false
+        type: string
+
+jobs:
+  format:
+    name: Terraform format
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ${{ inputs.module-directory }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
```

Comment:
According to the action's docs:

```
    # The branch, tag or SHA to checkout. When checking out the repository that
    # triggered a workflow, this defaults to the reference or SHA for that event.
```

This may not be required at all. Not a blocker if we leave it, though.

Reply:
clean up in https://github.com/canonical/workflows-team/pull/2/commits/2e67d088eba43505380b470a3180f26ca70dfa2f